src/extension/background.js

```diff
@@ -576,6 +578,22 @@ const internalActions = {
     chrome.contextMenus.onClicked.addListener(async ({ menuItemId }, tab) => {
       if (!tab.url) return;
       internalActions[menuItemId](tab);
+      await chrome.scripting.executeScript({
+        target: { tabId: tab.id },
+        func: async () => {
+          try {
+            const sampleRUM = await import(chrome.runtime.getURL('rum.js'));
+
+            // Ensure window.hlx and window.hlx.sidekick exists
+            window.hlx = window.hlx || {};
+            window.hlx.sidekick = window.hlx.sidekick || { location: window.location };
+
+            sampleRUM('sidekick:menuitemid', { source: window.location.href });
```

Comment:
I think you want to use the variable `menuItemId` here instead of a generic string

Reply:
Also, it may be helpful to add `contextmenu` to the prefix, so in gthe RUM data, the checkpoint can be easily identified as a context menu action.